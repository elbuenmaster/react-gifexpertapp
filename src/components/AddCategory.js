import React, { useState } from "react";
import Proptypes from 'prop-types'
export const AddCategory = ({ setCategorias }) => {
  
  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
    console.log('se cambio el valor de la cajita','{',e.target.value,'}');
  };

  const handleSubmit = (e) => {
              e.preventDefault();
              console.log("se dispara Submit");
              if (inputValue.trim().length > 2) {
                // setCategorias: se le agrega, usando una funcion CALLBACK que lea el valor del State y le aÃ±ada el nuevo valor
                setCategorias((cats) => [ inputValue,...cats]);
                setInputValue('');
                console.log("Submit exitoso");
                return true;
              }
              console.log("se cancela Submit");
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handleInputValue}></input>
    </form>
  );
};

AddCategory.propTypes={
    setCategorias: Proptypes.func.isRequired
}