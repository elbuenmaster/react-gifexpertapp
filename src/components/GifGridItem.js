import React from 'react'
import PropTypes from 'prop-types'
export const GifGridItem = ({id,title,url}) => {
    console.log(id)
    return (
        <div className="card animate__fadeIn">
        
           <img src={url} alt={title}></img> 
           <p>{title}</p>       
       
        </div>
    )
}

GifGridItem.propTypes={
    id:PropTypes.number.isRequired,
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired,
}
export default GifGridItem;