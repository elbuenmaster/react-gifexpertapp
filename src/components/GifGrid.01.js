import React, { useState, useEffect } from "react";
import { GifGridItem } from "../components/GifGridItem";
import {getGifs} from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const apiKey = "eBHH7BLGeQviyhM8FyoIaVveWFeN72yl";
  const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=10&api_key=${apiKey}`;

  const [images, setImages] = useState([]);

  

  useEffect(() => {
    getGifs(url)
    .then(imgs=>setImages(imgs));
  }, [url]);

  return (
    <>  
    <h3>{category}</h3>
    <div className='card-grid'>
    
      
        {images.map((imagen) => (
          <GifGridItem 
          key={imagen.id}
          {...imagen} />
        ))}
     
    </div>
   </> 
  );
};
