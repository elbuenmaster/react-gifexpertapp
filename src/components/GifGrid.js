import React, { useState, useEffect } from "react";
import { GifGridItem } from "../components/GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
 
  /* const [images, setImages] = useState([]);
  useEffect(() => {
    getGifs(url)
    .then(imgs=>setImages(imgs));
  }, [url]);
 */

  // al cargar la app por primera vez,data est√° vacio y loading esta en false
  const {data:images,loading}= useFetchGifs(category);
 
  return (
    <>  
    <h3>{category}</h3>
    {loading && <p>Cargando</p>}
   {   <div className='card-grid'>
    
      
        {images.map((imagen) => (
          <GifGridItem 
          key={imagen.id}
          {...imagen} />
        ))}
     
    </div> }
   </> 
  );
};

GifGrid.ProTypes={
  category:PropTypes.string.isRequired
}