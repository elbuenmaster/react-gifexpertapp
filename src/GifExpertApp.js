import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
export const GifExpertApp = () => {
  // const categories=['One Punch','Samurai X','Dragon Ball'];
const [categorias, setCategorias] = useState(["One Punch"]);

  /*   const handleAdd=()=>{
         console.log('agregar');
         setCategorias([...categorias,'Detective Conan']);
         setCategorias(cats=>[...cats,'Detective Conan'])
     } */

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory 
      setCategorias={setCategorias}></AddCategory>
      <hr />

      {categorias.map((cat) => {
        return <GifGrid key={cat} category={cat}></GifGrid>;
      })}
    </>
  );
};
