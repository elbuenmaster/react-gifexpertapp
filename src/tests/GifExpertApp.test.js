import React from 'react';
import {render} from '@testing-library/react';
import  '@testing-library/jest-dom';
import {shallow} from  'enzyme';
import {GifGrid} from '../components/GifGrid';
import {GifExpertApp} from '../GifExpertApp';

describe('testing GifExpertApp', () => {
    let wrapper;
     
    

    test('correcto snapshot del componente GifExpertAppw',()=>{

       
        let  wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();


        
    })


    test('debe mostrar una lista de categorias',()=>{
        const cats=['One Punch','Dragon Ball'];
        const wrapper = shallow(<GifExpertApp arreglodefault={cats}></GifExpertApp>);
        expect (wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(cats.length);
    });

})
