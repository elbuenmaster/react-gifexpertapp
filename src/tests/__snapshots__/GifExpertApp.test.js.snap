// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifExpertApp correcto snapshot del componente GifExpertAppw 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`testing GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;
